python/llm/src/bigdl/llm/transformers/speculative.py

```diff
@@ -195,6 +195,10 @@ def speculative_generate(self,
             # Draft model auto-regressively generate k tokens
             # Early stop when prob less then th_stop_draft
             for step_draft in range(max_step_draft):
+                if self.device.type == 'cpu':
+                    past_key_values = [
```

Comment:
Is this repeated every step, which may cause reductant latency and memory copy?

Reply:
could it be done once e.g. when init, and only process the additive part in the following?

Reply:
> could it be done once e.g. when init, and only process the additive part in the following?

Yes, we should try that.

Reply:
Is this implemented in the same when during our previous testing?